scripts/generating_yaml.py

```diff
@@ -0,0 +1,286 @@
+
+    def camera_to_serial_widget(camera_into_serial_port):
+        """Function used to create Jupyter widget.
+        It takes the parameter chosen from the widget and returns it such that it can be used as a variable.
+
+        Args:
+            camera_into_serial_port (str) : the port number of the Serial Adaptor that the camera is plugged into as selected from the widget.
+
+        Returns:
+            The result of the widget; the chosen serial port number
+
+        """
+        return camera_into_serial_port
+
+    def usbhub_sn_widget(USBhub_SN):
+        """Function used to create Jupyter widget.
+        It takes the parameter chosen from the widget and returns it such that it can be used as a variable.
+
+        Args:
+            USBhub_SN (str) : the serial number of the USB Hub as selected from the widget.
+
+        Returns:
+            The result of the widget; the chosen USB Hub serial number
+
+        """
+        return USBhub_SN
+
+    def camera_to_usb_widget(camera_into_USBhub_port):
+        """Function used to create Jupyter widget.
+        It takes the parameter chosen from the widget and returns it such that it can be used as a variable.
+
+        Args:
+            camera_into_USBhub_port (str) : the port number of the USB Hub that the camera is plugged into as selected from the widget.
+
+        Returns:
+            The result of the widget; the chosen USB port number
+
+        """
+        return camera_into_USBhub_port
+
+    birger_sn = data['birger_SN']
+    birger_serial_number = interactive(birger_sn_widget, birger_serial_number=birger_sn)
+    camera_sn = data['camera_SN']
+    camera_serial_number = interactive(camera_sn_widget, camera_serial_number=camera_sn)
+    lens_sn = data['lens_SN']
+    lens_serial_number = interactive(lens_sn_widget, lens_serial_number=lens_sn)
+    filter_ID = data['filter_ID']
+    filter_ID_code = interactive(filter_ID_widget, filter_ID_code=filter_ID)
+    serial_into_USBhub = data['serial_into_USBhub_port']
+    serial_into_USBhub_port = interactive(
+        serial_to_usb_widget, serial_into_USBhub_port=serial_into_USBhub)
+    camera_into_serial = data['camera_into_serial_port']
+    camera_into_serial_port = interactive(
+        camera_to_serial_widget, camera_into_serial_port=camera_into_serial)
+    USBhub = data['USBhub_SN']
+    USBhub_SN = interactive(usbhub_sn_widget, USBhub_SN=USBhub)
+    camera_into_USBhub = data['camera_into_USBhub_port']
+    camera_into_USBhub_port = interactive(
+        camera_to_usb_widget, camera_into_USBhub_port=camera_into_USBhub)
+
+    def add_device_widget(add_device):
+        """Function to add the details selected using the drop-down menu widgets to the 'data_list' dictionary.
+        The function is called by the widget and is then run when the user clicks on the widget button.
+
+        Args:
+            add_device : on clicking the widget, a device set is saved to the dict.
+
+        Returns:
```

Comment:
Maybe say `Output` like you do above instead of `Returns` as nothing is being returned.